### PR TITLE
Design java runtime query abstraction and implement OpenJ9Query

### PR DESCRIPTION
**Pull request by ShishirH**
Part of PRs which will implement runtime recommendations generation.

Signed-off-by: Shishir Halaharvi <shihala1@in.ibm.com>

### CODE REVIEW

**Comment by ShishirH** [comment]
@dinogun Can you please review this?

File: src/main/java/com/kruize/query/runtimes/java/openj9/OpenJ9JavaQuery.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+package com.kruize.query.runtimes.java.openj9;
+
+import com.kruize.query.runtimes.java.JavaQuery;
+
+public class OpenJ9JavaQuery extends JavaQuery
+{
```
Comment (dinogun) [inline src/main/java/com/kruize/query/runtimes/java/openj9/OpenJ9JavaQuery.java:22]: Same comment as earlier related to Prometheus queries structuring.

File: src/main/java/com/kruize/query/runtimes/java/JavaQuery.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ *******************************************************************************/
+
+package com.kruize.query.runtimes.java;
+
+public class JavaQuery
+{
+    /**
+     * @return String for a generic Java query that will fetch all applications exporting Java metrics
+     */
+    public String getAppsQuery()
```
[the remark this answers is not in the document]
Comment (ShishirH) [reply src/main/java/com/kruize/query/runtimes/java/JavaQuery.java:24]: Is `fetchJavaAppsQuery` alright?

File: src/main/java/com/kruize/query/runtimes/java/openj9/OpenJ9HeapQuery.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+package com.kruize.query.runtimes.java.openj9;
+
+import com.kruize.environment.DeploymentInfo;
+import com.kruize.exceptions.InvalidValueException;
+import com.kruize.query.runtimes.java.HeapQuery;
+
+public class OpenJ9HeapQuery implements HeapQuery
+{
+    private final String[] partsOfHeap = {"tenured-LOA", "tenured-SOA", "nursery-survivor", "nursery-allocate"};
```
[the remark this answers is not in the document]
Comment (ShishirH) [reply src/main/java/com/kruize/query/runtimes/java/openj9/OpenJ9HeapQuery.java:25]: I'm sorry, I didn't get you. What association are you talking about?

File: src/main/java/com/kruize/query/runtimes/java/JavaQuery.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (dinogun) [reply src/main/java/com/kruize/query/runtimes/java/JavaQuery.java:24]: Yup that sounds better

File: src/main/java/com/kruize/query/runtimes/java/openj9/OpenJ9HeapQuery.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (dinogun) [reply src/main/java/com/kruize/query/runtimes/java/openj9/OpenJ9HeapQuery.java:25]: The nomenclature that you have used, "tenured-LOA" are all related to the GENCON GC policy of OpenJ9. Balanced GC for example has a different heap structure. So the design of the HeapQuery has to be GC specific and has to ideally determine the GC policy first before you can decide which HeapQuery can be used.

See below for OpenJ9 GC reference.
https://www.eclipse.org/openj9/docs/gc/
https://developer.ibm.com/articles/garbage-collection-tradeoffs-and-tuning-with-openj9/

File: src/main/java/com/kruize/query/runtimes/java/HeapQuery.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
+/*******************************************************************************
+ * Copyright (c) 2019, 2019 IBM Corporation and others.
```
[the remark this answers is not in the document]
Comment (ShishirH) [reply src/main/java/com/kruize/query/runtimes/java/HeapQuery.java]: Changed year to 2020

File: src/main/java/com/kruize/query/runtimes/java/JavaQuery.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (ShishirH) [reply src/main/java/com/kruize/query/runtimes/java/JavaQuery.java]: Renamed `getAppsQuery` to `fetchJavaAppsQuery`

File: src/main/java/com/kruize/query/runtimes/java/openj9/OpenJ9HeapQuery.java [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+                return getNurseryAllocate(area, application);
+            default:
+                throw new InvalidValueException("No " + partOfHeap + " present in heap");
+        }
+    }
+
+    public String getTenuredLOA(String area, String name)
+    {
+        if (DeploymentInfo.getClusterType().toUpperCase().equals("DOCKER")) {
+            return "jvm_memory_" + area + "_bytes{area=\"heap\",id=\"tenured-LOA\"," +
```
[the remark this answers is not in the document]
Comment (ShishirH) [reply src/main/java/com/kruize/query/runtimes/java/openj9/OpenJ9HeapQuery.java]: Created a new package prometheus, and moved all prometheus query classes to it. Renamed classes accordingly to contain Prometheus.

**Pull request merged**